### DISCUSSION TITLE
Facing: error executing SQL statement [ERROR: role with OID 1126150 does not exist (28000)]

### DISCUSSION THREAD

**Discussion by sachin1kumar** [quarkiverse/quarkus-vault#241] [General]
**Complete stacktrace:**
org.hibernate.HibernateException - java.util.concurrent.CompletionException: org.hibernate.exception.GenericJDBCException: error executing SQL statement [ERROR: role with OID 1126150 does not exist (28000)] [select query]

Hiding the select query due to data privacy, when we run the query on OpenShift pod it works fine.

**Observation with this issue:**
Suppose currently we are facing this issue while using one of the endpoints of our microservice from Postman, making request multiple times but still same issue. Now, if we recreate the pod on Openshift then use the same endpoint we get the response and we will get the response till exact 1 hour and now if we use the same endpoint after one hour we will face the same issue again until we recreate the pod again on Openshift and this cycle repeats in order to make our endpoint works.

**Quarkus bom version:** 3.2.2.Final
**Quarkus vault version:** 'io.quarkiverse.vault:quarkus-vault:3.4.0'

**Comment by ruhilrinku** [comment]
Is there any update on the above mentioned issue ?

**Comment by kdubb** [comment]
@sachin1kumar Is this still an issue? Are you using rotating DB credentials? If so, have you set the `max-lifetime` to recycle connections faster than your Vault lease TTL?

**Comment by sachin1kumar** [comment]
Yes, this is still an issue. Yes, we are using rotating DB credentials.

**Comment by kdubb** [comment]
And you have `max-lifetime` set?

**Comment by sachin1kumar** [comment]
We tried that as well but it didn't work out.

Kind regards,

Sachin Kumar

**Comment by vsevel** [comment]
could you provide a reproducer?
